app/controllers/ArrivalMovementController.scala

```diff
@@ -64,4 +68,17 @@ class ArrivalMovementController @Inject()(cc: ControllerComponents,
         }
       }
   }
-}
+
+  def getArrival(arrivalId: String): Action[AnyContent] =
+    (authAction andThen validateAcceptJsonHeaderAction).async {
+    implicit request => {
+      arrivalConnector.get(arrivalId).map { r =>
+          r match {
+            case Right(a) => Ok(Json.toJson(ResponseArrival(a)))
+            case Left(response) => handleNon2xx(response)
```

Comment:
Can we use long names for the variables? <br><br>s/r/response<br>s/a/arrival<br><br>invalidResponse for the Left.